buildSrc/src/main/kotlin/tools-integrations-conventions.gradle.kts

```diff
@@ -51,51 +47,31 @@ configurations.all {
         val icebergVersionToUse = System.getProperty("nessie.versionIceberg")
         val req = requested
         if (req is ModuleComponentSelector) {
-          var moduleComponentSelector = req as DefaultModuleComponentSelector
           if (
             icebergVersionToUse != null &&
               req.group == "org.apache.iceberg" &&
               (req.version.isEmpty() || req.version != icebergVersionToUse)
           ) {
-            // TODO get rid of the internal Gradle classes DefaultImmutableVersionConstraint +
-            //  DefaultModuleComponentSelector.
-            val version = DefaultImmutableVersionConstraint.of(icebergVersionToUse)
-            val target =
-              DefaultModuleComponentSelector.newSelector(
-                req.moduleIdentifier,
-                version,
-                req.attributes,
-                moduleComponentSelector.getCapabilitySelectors()
-              )
-            useTarget(target, "Managed Iceberg version to $version (attributes: ${req.attributes})")
+            useTarget(
+              "${req.group}:${req.module}:$icebergVersionToUse",
+              "Managed Iceberg version to $version (attributes: ${req.attributes})"
```

Comment:
does it matter that we seem to no longer pass the `req.attributes` ?

Reply:
no, we're good here